### PR TITLE
dd-63-refactor-code

### PR DESCRIPTION
**Pull request by RykovaMariia**
# Task

[Refactor Code](https://divas801801.atlassian.net/jira/software/projects/DD/boards/2?selectedIssue=DD-63)

# Description

- delete TODO and other comments
- move location services to Admin feature
- refactor interfaces
- refactor utils

## Pull request type

Please check the type of change your PR introduces:

- [ ] 🍕 Feature
- [ ] 🐛 Bugfix
- [x] 🧑‍💻 Code Refactor
- [ ] 📝 Documentation Update
- [ ] 🤖 Build related changes
- [ ] 🔥 Performance Improvements
- [ ] Other (please describe):

# Demo (situationally)

# Has It Been Tested?

- [ ] Yes
- [ ] No, and this is why: please replace this line with details on why tests have not been included
- [x] Tests are not needed for this task
- [ ] I need help with writing tests

### CODE REVIEW

**Review by YuliyaVoronovich — DISMISSED** [review]

File: src/app/features/search/components/search-filter/search-filter.component.ts [hunk 1]
```diff
@@ -2,12 +2,29 @@ import { NgClass } from '@angular/common';
...
+  if (width < mobileWidth) {
+    return 2;
+  }
+  if (width < middleWidth) {
+    return 3;
+  }
+  if (width < tabletWidth) {
+    return 4;
+  }
+  return 5;
```
Comment (ChickenMcNuggetss) [inline src/app/features/search/components/search-filter/search-filter.component.ts:25]: Magic numbers

Comment (RykovaMariia) [reply src/app/features/search/components/search-filter/search-filter.component.ts:25]: my function is called getItemsCountFromWidth, so 2, 3, 4, 5 are ItemsCount

Comment (ChickenMcNuggetss) [reply src/app/features/search/components/search-filter/search-filter.component.ts:25]: What kind of items?

Comment (RykovaMariia) [reply src/app/features/search/components/search-filter/search-filter.component.ts:25]: That's another question :smile:

File: src/app/features/orders/components/order-card/order-card.component.ts [hunk 2]
```diff
@@ -84,7 +81,7 @@ export class OrderCardComponent {
 
   public duration() {
     if (!this.segments().length) return '';
-    return calculateStopDuration(this.segments()[this.segments().length - 1].time[1], this.segments()[0].time[0]);
+    return calculateTrainStopDuration(this.segments()[this.segments().length - 1].time[1], this.segments()[0].time[0]);
```
Comment (Dewlar) [inline src/app/features/orders/components/order-card/order-card.component.ts]: sample variable name: `departureDate/Time`
```suggestion
    const departureDate = this.segments()[0].time[0];
    const arrivalDate = this.segments()[this.segments().length - 1].time[1];
    return calculateTrainStopDuration(arrivalDate, departureDate);
```

File: src/app/features/search/components/search-form/search-form.component.ts [hunk 3]
```diff
@@ -85,12 +85,12 @@ export class SearchFormComponent implements OnInit {
   public ngOnInit() {
     const { from, to } = this.searchForm.controls;
```
Comment (Dewlar) [inline src/app/features/search/components/search-form/search-form.component.ts]: we use a getter for `controls` so we can use it here ?
```suggestion
    const { from, to } = this.controls;
```

File: src/app/features/orders/interfaces/user.interface.ts [hunk 4]
```diff
@@ -1,6 +1,8 @@
+import { Role } from '@shared/enums/role.enum';
+
 export interface User {
   id: number;
   email: string;
   name: string;
-  role: 'manager' | 'user';
+  role: keyof typeof Role;
```
Comment (Dewlar) [inline src/app/features/orders/interfaces/user.interface.ts:7]: I like this use ❤️

File: src/app/features/admin/validators/connected-station.validator.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { StationsService } from '@core/services/stations/stations.service';
+
+export function connectedStationValidator(stationsService: StationsService): ValidatorFn {
```
Comment (Dewlar) [inline src/app/features/admin/validators/connected-station.validator.ts:4]: oh, i didn't know you could pass a service to the validator. thanks for the idea ❤️

Comment (RykovaMariia) [reply src/app/features/admin/validators/connected-station.validator.ts:4]: I'm not sure if this is the right thing

**Review by Dewlar — APPROVED** [review]
finally we did it

**Review by YuliyaVoronovich — APPROVED** [review]

**Pull request merged**